src/coreclr/jit/llvm.cpp

```diff
@@ -2392,6 +2458,48 @@ void Llvm::PlaceAndConvertShadowStackLocals()
     lowerToShadowStack();
 }
 
+static bool placeLocalInAlloca(LclVarDsc* varDsc)
+{
+    return !varDsc->lvInSsa;
+}
+
+void Llvm::createAllocasForLocalsWithAddrOp()
+{
+    m_allocas = new std::vector<Value*>(_compiler->lvaCount, nullptr);
+
+    for (unsigned lclNum = 0; lclNum < _compiler->lvaCount; lclNum++)
+    {
+        LclVarDsc* varDsc = _compiler->lvaGetDesc(lclNum);
+
+        // TODO-LLVM LCLBLK - outgoing arg space
+        if (varDsc->lvType != TYP_LCLBLK && canStoreLocalOnLlvmStack(varDsc) && placeLocalInAlloca(varDsc))
+        {
+            CORINFO_CLASS_HANDLE classHandle = NO_CLASS_HANDLE;
+            if (varDsc->lvType == TYP_STRUCT)
+            {
+                ClassLayout* layout = varDsc->GetLayout();
+                classHandle         = layout->GetClassHandle();
+            }
+            Type* llvmType = getLlvmTypeForCorInfoType(toCorInfoType(varDsc->lvType), classHandle);
+            if (llvmType->isIntegerTy() && llvmType->getIntegerBitWidth() < 32)
+            {
+                llvmType = Type::getInt32Ty(_llvmContext);
+            }
+            Value* allocaValue = _prologBuilder.CreateAlloca(llvmType);
+            (*m_allocas)[lclNum] = allocaValue;
+
+            if (varDsc->lvIsParam)
+            {
+                LlvmArgInfo argInfo = getLlvmArgInfoForArgIx(lclNum);
+                assert(argInfo.m_argIx >= 0); // check this arg is not on the shadow stack
```

Comment:
A small helper could be added to `LlvmArgInfo` that checks for this, say `IsLlvmArg()`.

Reply:
Added.